src/coreComponents/dataRepository/BufferOps_inline.hpp

```diff
@@ -784,6 +784,60 @@ localIndex Unpack( buffer_unit_type const * & buffer,
   return sizeOfUnpackedChars;
 }
 
+template< bool DO_PACKING >
+localIndex Pack( buffer_unit_type * & buffer,
+                 SortedArrayView< localIndex const > const & var,
+                 arrayView1d< localIndex const > const & packList,
+                 arraySlice1d< globalIndex const > const & localToGlobal )
+{
+
+  localIndex length = 0;
+  array1d< localIndex > temp( var.size() );
+
+  for( auto a : packList )
+  {
+    if( var.count( a ) )
+    {
+      temp[length] = a;
+      ++length;
+    }
+  }
+  temp.resize( length );
+  localIndex sizeOfPackedChars = Pack< DO_PACKING >( buffer, length );
+
+
+  for( localIndex a=0; a<length; ++a )
+  {
+    sizeOfPackedChars += Pack< DO_PACKING >( buffer, localToGlobal[temp[a]] );
+  }
+
+  return sizeOfPackedChars;
+}
+
+//template< typename SORTED >
+//localIndex Unpack( buffer_unit_type * & buffer,
+//                   SortedArrayView< localIndex const > const & var,
+//                   arrayView1d< localIndex const > const & packList,
```

Comment:
I wondered if I needed to filter based on the packList but I don't see why it would be necessary when unpacking.

Reply:
That sounds right to me, you're just unpacking global indices and you don't care where they came from.

Reply:
But if that's the case remove the commented out code.